qemu/tests/virtio_fs_map_uid_gid.py

```diff
@@ -0,0 +1,257 @@
+        cmd_run_virtiofsd = params["cmd_run_virtiofsd"] % sock_path
+        cmd_run_virtiofsd += " --shared-dir %s" % shared_dir
+        cmd_run_virtiofsd += fs_binary_extra_options
+        error_context.context("Running daemon command %s with %s user." %
+                              (cmd_run_virtiofsd, user_name),
+                              test.log.info)
+        p_vfsd = aexpect.ShellSession("runuser -l " + user_name +
+                                      " -c '" + cmd_run_virtiofsd + "'",
+                                      auto_close=False,
+                                      output_func=utils_misc.log_line,
+                                      output_params=(vfsd_log_name,),
+                                      prompt=r"^\[.*\][\#\$]\s*$")
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+
+        vm = env.get_vm(params.get("main_vm"))
+        vm.verify_alive()
+        session = vm.wait_for_login()
+
+        if not utils_misc.make_dirs(fs_dest, session):
+            test.fail("Creating directory was failed!")
+        error_context.context("Mount virtiofs target %s to %s inside"
+                              " guest." % (fs_target, fs_dest),
+                              test.log.info)
+        if not utils_disk.mount(fs_target, fs_dest, 'virtiofs', session=session):
+            test.fail('Mount virtiofs target failed.')
+
+        for guest_user in ["root", common_guest_user]:
+            # create a new file in guest.
+            file_name = create_file_in_guest(guest_user)
+            # if map type is 1 to 1, then create file will fail with
+            # a common user, so there is no need to check the uid/gid
+            if not (map_type == "one_to_one"
+                    and guest_user == common_guest_user):
+                uid_gid_guest = get_file_owner_id_guest(file_name)
+                uid_gid_host = get_file_owner_id_host(file_name)
+                expect_id = get_expect_user_id(guest_user)
+                msg = "The new file's uid:gid is wrong from %s\n"
+                msg += "expect one is %s\nthe real one is %s."
+                if uid_gid_guest != expect_id["guest"]:
+                    test.fail(msg % ("guest", expect_id["guest"], uid_gid_guest))
+                if uid_gid_host != expect_id["host"]:
+                    test.fail(msg % ("host", expect_id["host"], uid_gid_host))
+    finally:
+        error_context.context("Clean the env, delete the user on guest.",
+                              test.log.info)
+        utils_disk.umount(fs_target, fs_dest, 'virtiofs', session=session)
+        utils_misc.safe_rmdir(fs_dest, session=session)
+        if session.cmd_status("id %s" % common_guest_user) == 0:
+            session.cmd(del_user_cmd % common_guest_user)
+        if vm and vm.is_alive():
+            vm.destroy()
+        error_context.context("Delete the user on host.", test.log.info)
+        p_vfsd.kill()
```

Comment:
Same to the above: should check if the p_vfsd is created and is not `None` before `p_vfsd.kill()`